bibtexparser/bibdatabase.py

```diff
@@ -101,6 +102,10 @@ def expand_string(self, name):
             return BibDataStringExpression.expand_if_expression(
                 self.strings[name])
         except KeyError:
+            if self.string_provider:
+                val = self.string_provider(name)
+                if val:
```

Comment:
Here we want `val is not None` to allow empty string values. An alternative would be to expect the string provider to raise a `KeyError` or something similar. I think it is fine to allow both by making using the `is` as suggested.